<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>349</width>
    <height>615</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>UpWind::NMEA Lightweight Client</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QFrame.headerFrame {
border: none;
border-top-left-radius: 4px;
border-top-right-radius: 4px;
background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
stop: 0.01 rgba(120, 120, 120),
stop: 0.05 rgba(150, 150, 150),
stop: 0.5 rgba(120, 120, 120),
stop: 0.52 rgba(80, 80, 80),
stop: 0.9 rgba(50, 50, 50),
stop: 1.0 rgba(50, 50, 50));
}

QFrame.headerFrame QLabel {
border: none;
background: transparent;
font-weight: bold;
color: rgba(225, 225, 225);
}

QFrame.containerFrame { border-top: 1px solid rgba(100, 100, 100);
border-left: 1px solid rgba(100, 100, 100);
border-right: 1px solid rgba(100, 100, 100);
border-bottom: 1px solid rgba(75, 75, 75);
padding: 3px;
background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
stop: 0.0 rgba(95, 95, 95),
stop: 0.15 rgba(75, 75, 75),
stop: 0.9 rgba(95, 95, 95),
stop: 1.0 rgba(45, 45, 45));
border-top-left-radius: 8px;
border-top-right-radius: 8px;
border-bottom-left-radius: 4px;
border-bottom-right-radius: 4px;
}

QFrame.containerFrame QLabel {
border: none;
background: transparent;
font-weight: bold;
color: rgba(225, 225, 225);
}

QFrame.containerFrame QPushButton {
padding: 4px;
border-radius: 4px;
border: 1px solid rgba(125, 125, 125);
background: qlineargradient(x1:0, y1:0, x2: 0, y2:1,
stop: 0.0 rgba(25, 25, 25),
stop: 0.05 rgba(75, 75, 75),
stop: 0.15 rgba(125, 125, 125),
stop: 0.85 rgba(75, 75, 75),
stop: 1.0 rgba(25, 25, 25));
color: rgba(225, 225, 225);
}

QFrame.containerFrame QPushButton:pressed {
background:qlineargradient(x1:0, y1:0, x2: 0, y2:1,
stop: 0.0 rgba(0, 0, 0),
stop: 0.05 rgba(50, 50, 50),
stop: 0.15 rgba(100, 100, 100),
stop: 0.85 rgba(50, 50, 50),
stop: 1.0 rgba(0, 0, 0));
color: rgba(200, 200, 200);
}

QFrame.containerFrame QPushButton:disabled {
background:qlineargradient(x1:0, y1:0, x2: 0, y2:1,
stop: 0.0 rgba(50, 50, 50),
stop: 0.05 rgba(100, 100, 100),
stop: 0.15 rgba(150, 150, 150),
stop: 0.85 rgba(100, 100, 100),
stop: 1.0 rgba(50, 50, 50));
}

QFrame.contentFrame {
border-top-left-radius: 0px;
border-top-right-radius: 0px;
border-bottom-left-radius: 2px;
border-bottom-right-radius: 2px;
border-right: 1px solid rgba(125, 125, 125);
border-left: 1px solid rgba(125, 125, 125);
background: qlineargradient(x1:0, y1:0, x2:0, y2:1,
stop: 0.0 rgba(125, 125, 125),
stop: 0.02 rgba(200, 200, 200),
stop: 0.98 rgba(200, 200, 200),
stop: 1.0 rgba(125, 125, 125));
}

QFrame.contentFrame QLabel {
color: rgba(35, 35, 35);
font-weight: normal;
background: transparent;
border: none;
}

QFrame.contentFrame QCheckBox {
background: transparent;
border: none;
}

QFrame.contentFrame QPushButton {
border-radius: 4px;
border: 1px solid rgba(75, 75, 75);
background: qlineargradient(x1:0, y1:0, x2: 0, y2:1,
stop: 0.0 rgba(25, 25, 25),
stop: 0.05 rgba(75, 75, 75),
stop: 0.15 rgba(125, 125, 125),
stop: 0.85 rgba(75, 75, 75),
stop: 1.0 rgba(25, 25, 25));
color: rgba(225, 225, 225);
}

QFrame.contentFrame QPushButton:pressed {
background:qlineargradient(x1:0, y1:0, x2: 0, y2:1,
stop: 0.0 rgba(0, 0, 0),
stop: 0.05 rgba(50, 50, 50),
stop: 0.15 rgba(100, 100, 100),
stop: 0.85 rgba(50, 50, 50),
stop: 1.0 rgba(0, 0, 0));
color: rgba(200, 200, 200);
}

QTextEdit {
border: 1px solid rgba(175, 175, 175);
background: rgba(255, 255, 255);
border-top-left-radius: 0px;
border-top-right-radius: 0px;
border-bottom-right-radius: 4px;
border-bottom-left-radius: 4px;
}
</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="styleSheet">
    <string notr="true">QWidget #centralWidget {
background: rgba(225, 225, 225);
}</string>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <property name="verticalSpacing">
     <number>0</number>
    </property>
    <item row="3" column="0" colspan="3">
     <widget class="QTextEdit" name="textOutput">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>200</height>
       </size>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="9" column="0" colspan="3">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="clearButton">
        <property name="text">
         <string>Clear</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item row="2" column="0" colspan="3">
     <widget class="QFrame" name="frame">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>35</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <property name="class" stdset="0">
       <string>headerFrame</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>NMEA data</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="0" colspan="3">
     <widget class="QFrame" name="frame_2">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>120</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="margin">
        <number>0</number>
       </property>
       <item>
        <widget class="QFrame" name="frame_3">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout">
          <property name="spacing">
           <number>0</number>
          </property>
          <property name="margin">
           <number>0</number>
          </property>
          <item>
           <widget class="QFrame" name="serverHeader">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>35</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>35</height>
             </size>
            </property>
            <property name="sizeIncrement">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <property name="class" stdset="0">
             <string>headerFrame</string>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QLabel" name="label_5">
               <property name="text">
                <string>Server settings</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame_5">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <property name="class" stdset="0">
             <string>contentFrame</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_4">
             <item row="2" column="1">
              <widget class="QLineEdit" name="ipEdit">
               <property name="text">
                <string>127.0.0.1</string>
               </property>
              </widget>
             </item>
             <item row="2" column="2">
              <widget class="QLineEdit" name="portEdit">
               <property name="text">
                <string>9999</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLabel" name="label">
               <property name="text">
                <string>Server address:</string>
               </property>
              </widget>
             </item>
             <item row="1" column="2">
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>Server port:</string>
               </property>
              </widget>
             </item>
             <item row="3" column="0" colspan="2">
              <widget class="QPushButton" name="connectBtn">
               <property name="minimumSize">
                <size>
                 <width>60</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>Connect</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="frame_4">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <property name="spacing">
           <number>0</number>
          </property>
          <property name="margin">
           <number>0</number>
          </property>
          <item>
           <widget class="QFrame" name="serverHeader_2">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>35</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>35</height>
             </size>
            </property>
            <property name="sizeIncrement">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <property name="class" stdset="0">
             <string>headerFrame</string>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QLabel" name="label_6">
               <property name="text">
                <string>Serial settings</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame_6">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <property name="class" stdset="0">
             <string>contentFrame</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_3">
             <item row="0" column="0" colspan="2">
              <widget class="QCheckBox" name="serialCheckBox">
               <property name="text">
                <string>Enable serial port output</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QSpinBox" name="serialSpinBox">
               <property name="maximum">
                <number>99</number>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>Serial port</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>connectBtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>makeConnection()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>296</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>344</x>
     <y>63</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>makeConnection()</slot>
 </slots>
</ui>
